addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -622,6 +626,66 @@ options.registry.Theme = options.Class.extend({
 
         await this._reloadBundles();
     },
+    /**
+     * @see this.selectClass for parameters
+     */
+    async openCustomCodeDialog(previewMode, widgetValue, params) {
+        let websiteId;
+        let aceEditor;
+
+        let fieldName, contentText;
+        if (widgetValue === 'head') {
+            fieldName = 'custom_code_head';
+            contentText = _t('Enter code that will be added into the <head> of every page of your site.');
+        } else {
+            fieldName = 'custom_code_footer';
+            contentText = _t('Enter code that will be added before the </body> of every page of your site.');
+        }
+
+        this.trigger_up('context_get', {
+            callback: (ctx) => {
+                websiteId = ctx['website_id'];
+            },
+        });
+        const websites = await this._rpc({
+            model: 'website',
+            method: 'read',
+            args: [[websiteId], ['custom_code_head', 'custom_code_footer']],
+        });
+
+        const $content = $(core.qweb.render('website.custom_code_dialog_content', {
+            text: contentText,
+        }));
+
+        const dialog = new Dialog(this, {
+            title: widgetValue,
+            $content,
+            buttons: [
+                {
+                    text: _t("Save"),
+                    classes: 'btn-primary',
+                    click: async () => {
+                        await this._rpc({
+                            model: 'website',
+                            method: 'write',
+                            args: [
+                                [websiteId],
+                                {[fieldName]: aceEditor && aceEditor.getValue()},
```

Comment:
- I am not sure why you define the aceEditor variable empty above, open the dialog, then initialize the variable?
- The promise of the option method `openCustomCodeDialog` should only be resolved when the user has finished dealing with the modal. So I guess you should wrap the modal opening in a Promise. This is to be sure that options do not run in parallel.